shell/common/gin_helper/destroyable.cc

```diff
@@ -13,15 +12,9 @@ namespace gin_helper {
 
 namespace {
 
-v8::Global<v8::FunctionTemplate>* GetDestroyFunc() {
-  static base::NoDestructor<v8::Global<v8::FunctionTemplate>> destroy_func;
-  return destroy_func.get();
-}
+v8::Eternal<v8::FunctionTemplate> destroy_func_;
```

Comment:
Hm, I know this isn't a change in _this_ PR but this does highlight that this code is not multi-context safe 🤔 This part in particular as it lives in `//common` if used in the renderer (which it currently isn't) would be a security hole if consumed as the destroy functions would be made in one context and used / exposed in another.

I think we should either:
* Remove usages of `v8::Eternal` in `//common` and `//renderer`
* Or ensure that they `Eternal` objects are attached to a specific context so that multi-context scenarios work.

Reply:
@MarshallOfSound just to confirm -- since this PR doesn't change the risk compared to the currently-shipping code, it sounds like you're saying that isolation/removal would be a followup task, correct?

Reply:
I think we could move `gin_helper::Destroyable` from `//common` into `//browser`.

There is a helper function `gin_helper::Destroyable::IsDestroyed()` that is used by `common/gin_helper/function_template.h`, but that helper isn't coupled with the Eternal/Global templates and could stay in `//common` as a standalone.

@MarshallOfSound if that sounds correct, I can do that as a followup

Reply:
Oh yeah, happy to be a separate item. Just noting it for future work, it's definitely unsafe in it's current form and actually is a memory leak in renderers 🤔

Reply:
Thinking about it some more `Eternal` isn't safe in `//browser` either because of node worker threads 🤔 We should make it context safe or nuke it

Reply:
@MarshallOfSound would PerIsolateData be an appropriate way to handle this?

Reply:
Unfortunately not because the same isolate can be used by multiple contexts.